django-backend/fecfiler/scha_transactions/serializers.py

```diff
@@ -150,6 +158,20 @@ def create_or_update_contact(self, validated_data: dict):
         else:
             Contact.objects.filter(id=tran_contact_id).update(**contact_data)
 
+    def create_or_update_memo_text(self, validated_data: dict):
+        memo_data = validated_data.pop("memo_text", None)
+        tran_memo_text_id = validated_data.get("memo_text_id", None)
+        if not tran_memo_text_id:
+            if not memo_data:
+                raise ValidationError(
+                    {"memo_text": ["No memo_text or memo_text_id provided"]}
```

Comment:
Unlike Contacts, I think we want to allow transactions to be saved if they don't have a memo_text attached.  it should be optional in many cases

Reply:
The front-end always sends a value for memo_text, it's just that the text4000 field is sometimes empty.  This is because we don't want to create a new transaction without creating a memo_text record.  I was on the fence about that, but ultimately I don't think we want to be deleting records every time someone edits a memo to empty.

Reply:
I see what you're getting at with clearing out the field.  I think the cost is pretty high for simplifying that case.  I think the edge case where they clear it out can be left alone and optimized to where we delete empty ones in the future.

There are pretty big ramifications to having a memo_text record for every transaction.  One is that the .fec will have memo_text record per transaction, which will be almost on the order of doubling its size (if memo's are rare).  Disclosures (pdf versions) will have a lot of empty pages cluttering the document.  We'll have a whole lot more records in the db than what we need.